src/contract_wrappers/tokens/security_token_wrapper.ts

```diff
@@ -1048,19 +1056,19 @@ export default class SecurityTokenWrapper extends ERC20TokenWrapper {
           dateToBigNumber((params.data as USDTieredSTOData).startTime).toNumber(),
           dateToBigNumber((params.data as USDTieredSTOData).endTime).toNumber(),
           (params.data as USDTieredSTOData).ratePerTier.map(e => {
-            return e.toNumber();
+            return valueToWei(e, decimals).toString();
           }),
           (params.data as USDTieredSTOData).ratePerTierDiscountPoly.map(e => {
-            return e.toNumber();
+            return valueToWei(e, decimals).toString();
           }),
           (params.data as USDTieredSTOData).tokensPerTierTotal.map(e => {
-            return e.toNumber();
+            return valueToWei(e, decimals).toString();
           }),
           (params.data as USDTieredSTOData).tokensPerTierDiscountPoly.map(e => {
-            return e.toNumber();
+            return valueToWei(e, decimals).toString();
           }),
-          (params.data as USDTieredSTOData).nonAccreditedLimitUSD.toNumber(),
-          (params.data as USDTieredSTOData).minimumInvestmentUSD.toNumber(),
+          valueToWei((params.data as USDTieredSTOData).nonAccreditedLimitUSD, decimals).toString(),
+          valueToWei((params.data as USDTieredSTOData).minimumInvestmentUSD, decimals).toString(),
```

Comment:
USD amounts are always expressed with 18 decimals, they don't depend on the amount of security token decimals